sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -597,6 +619,107 @@ private[hive] class HiveMetastoreCatalog(val client: HiveClient, hive: HiveConte
     }
   }
 
+  private def convertToOrcRelation(metastoreRelation: MetastoreRelation): LogicalRelation = {
```

Comment:
How much of this is duplicated with `convertToParquetRelation`?  It would be awesome if we could just make this `ConvertToHadoopFSRelation` and handle all the cases (or just ORC/Parquet through a common path).  I'd have to look closer, but I think its likely that the parquet specific things (like parquetOption with the merged schema) are actually not required anymore.

The fact that this code path is parquet specific predates a common interface for reading files in Spark SQL.